generator/overrides/gen.go

```diff
@@ -0,0 +1,435 @@
+	Omit bool `marker:",optional"`
+	// OmmitInPlugin indicates that this field cannot be overriden in a devfile plugin.
+	OmitInPlugin bool `marker:",optional"`
+	// Description indicates the description that should be added as Go documentation on the generated structs.
+	Description string `marker:",optional"`
+}
+
+var (
+	overridesFieldMarker = markers.Must(markers.MakeDefinition("devfile:overrides:include", markers.DescribesField, FieldOverridesInclude{}))
+	overridesTypeMarker  = markers.Must(markers.MakeDefinition("devfile:overrides:generate", markers.DescribesType, struct{}{}))
+)
+
+// +controllertools:marker:generateHelp
+
+// Generator generates additional GO code for the overriding of elements in devfile parent or plugins.
+type Generator struct {
+
+	// IsForPluginOverrides indicates that the generated code should be done for plugin overrides.
+	// When false, the parent overrides are generated
+	IsForPluginOverrides bool `marker:"isForPluginOverrides,optional"`
+
+	packageTypes      map[string]*markers.TypeInfo
+	suffix            string
+	rootTypeToProcess typeToProcess
+}
+
+// RegisterMarkers registers the markers of the Generator
+func (Generator) RegisterMarkers(into *markers.Registry) error {
+	if err := markers.RegisterAll(into, overridesFieldMarker, overridesTypeMarker); err != nil {
+		return err
+	}
+	into.AddHelp(overridesFieldMarker, FieldOverridesInclude{}.Help())
+	into.AddHelp(overridesTypeMarker, markers.SimpleHelp("Overrides", "indicates that a type should be selected to create Overrides for it"))
+	return genutils.RegisterUnionMarkers(into)
+}
+
+// Generate generates the artifacts
+func (g Generator) Generate(ctx *genall.GenerationContext) error {
+	for _, root := range ctx.Roots {
+		ctx.Checker.Check(root, func(node ast.Node) bool {
+			// ignore interfaces
+			_, isIface := node.(*ast.InterfaceType)
+			return !isIface
+		})
+
+		root.NeedTypesInfo()
+
+		var rootStructToOverride *markers.TypeInfo
+		g.packageTypes = map[string]*markers.TypeInfo{}
+		if err := markers.EachType(ctx.Collector, root, func(info *markers.TypeInfo) {
+			if info.Markers.Get(overridesTypeMarker.Name) != nil {
+				if rootStructToOverride == nil {
+					rootStructToOverride = info
+				} else {
+					root.AddError(fmt.Errorf("Marker %v should be added to only one Struct type, but was added on %v and %v",
+						overridesTypeMarker.Name,
+						rootStructToOverride.Name,
+						info.Name,
+					))
+				}
+			}
+			g.packageTypes[info.RawSpec.Name.Name] = info
+		}); err != nil {
+			root.AddError(err)
+			return nil
+		}
+
+		if rootStructToOverride == nil {
+			root.AddError(fmt.Errorf("Marker %v should be added to at least one Struct type",
+				overridesTypeMarker.Name,
+			))
+			return nil
+		}
+
+		config := printer.Config{
+			Tabwidth: 2,
+			Mode:     printer.UseSpaces,
+		}
+
+		buf := new(bytes.Buffer)
+		buf.WriteString("package " + root.Name)
+		buf.WriteString("\n\n")
```

Comment:
Probably because it's the first one I worked on :-)

Reply:
I used the new function extracted to `genutils`, so that the approach is the same for all GO-source-related generators.

Fixed in commit https://github.com/devfile/api/pull/133/commits/37c1e341bb7bb541566dbf0b63a536b64e1b6028